Repository: plone.scale


Branch: refs/heads/master
Date: 2016-02-03T15:44:19-08:00
Author: Fulvio Casali (fulv) <fulviocasali@gmail.com>
Commit: https://github.com/plone/plone.scale/commit/3e5f2db8a57d011bf9b660912343a9f9a3d26da7

Fixes Products.CMFPlone#1361

Files changed:
M plone/scale/storage.py

diff --git a/plone/scale/storage.py b/plone/scale/storage.py
index 986ed20..2135f6f 100644
--- a/plone/scale/storage.py
+++ b/plone/scale/storage.py
@@ -182,7 +182,7 @@ def _cleanup(self):
             if isinstance(key, tuple):
                 del storage[key]
             # clear cache from scales older than one day
-            if (modified_time and
+            elif (modified_time and
                     value['modified'] < modified_time - KEEP_SCALE_MILLIS):
                 del storage[key]
 


Repository: plone.scale


Branch: refs/heads/master
Date: 2016-02-04T09:02:01-08:00
Author: Fulvio Casali (fulv) <fulviocasali@gmail.com>
Commit: https://github.com/plone/plone.scale/commit/7c4c41724db459ade2118eb1e38bb66373438d8f

Add changelog.

Files changed:
M docs/changes.rst

diff --git a/docs/changes.rst b/docs/changes.rst
index c3439b6..f4e2f48 100644
--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -5,7 +5,9 @@ Changelog
 1.4.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Fix KeyError in storage.AnnotationStorage._cleanup when attempting
+  to delete the storage for the same key twice.
+  [fulv]
 
 
 1.4 (2015-12-07)


Repository: plone.scale


Branch: refs/heads/master
Date: 2016-02-04T18:35:20+01:00
Author: Jens W. Klein (jensens) <jk@kleinundpartner.at>
Commit: https://github.com/plone/plone.scale/commit/78c72ad2ac45c64a8f2350a264749c054b28618d

Merge pull request #8 from plone/fulv-issue1361

Fixes Products.CMFPlone#1361

Files changed:
M docs/changes.rst
M plone/scale/storage.py

diff --git a/docs/changes.rst b/docs/changes.rst
index c3439b6..f4e2f48 100644
--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -5,7 +5,9 @@ Changelog
 1.4.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Fix KeyError in storage.AnnotationStorage._cleanup when attempting
+  to delete the storage for the same key twice.
+  [fulv]
 
 
 1.4 (2015-12-07)
diff --git a/plone/scale/storage.py b/plone/scale/storage.py
index 986ed20..2135f6f 100644
--- a/plone/scale/storage.py
+++ b/plone/scale/storage.py
@@ -182,7 +182,7 @@ def _cleanup(self):
             if isinstance(key, tuple):
                 del storage[key]
             # clear cache from scales older than one day
-            if (modified_time and
+            elif (modified_time and
                     value['modified'] < modified_time - KEEP_SCALE_MILLIS):
                 del storage[key]
 


